### ISSUE TITLE
[Feature Request] Calling `.eval` on criterion

### ISSUE THREAD

**Issue by YannDubs** [skorch-dev/skorch#565]
usually criterions in pytorch inherit from nn.Module, meaning that they should have a `self.training` option that says whether it is during train or validation. yet skorch does not update it. This is useful for cases where there are multiple losses and the weights of each loss change during training (but not testing). E.g. beta-vae with annealing, Gans .... It seems that you though about that as `get_loss` takes `training` as argument. It would be useful to update the criterion accordingly :

```
def get_loss(self, y_pred, y_true, X=None, training=False):
    """Return the loss for this batch. [...]"""
    y_true = to_tensor(y_true, device=self.device)

    if isinstance(self.criterion_, nn.Module):
        if training:
            self.criterion_.train()
        else:
            self.criterion_.eval()

    return self.criterion_(y_pred, y_true)
```

**Comment by BenjaminBossan** [comment]
This sounds like a good idea. If you want to, go ahead and implement it. A minor thing I would change: use `self.criterion_.train(training)` to be consistent with the rest of skorch.

**Comment by BenjaminBossan** [comment]
@YannDubs Do you still plan to work on this?

**Comment by YannDubs** [comment]
Yes but after ICML deadline (7th Feb).

**Comment by BenjaminBossan** [comment]
@YannDubs Are you still willing to work on this?
